### ISSUE TITLE
Calling tweet_detail on a tweet with limited replies/visibility results in an error

### ISSUE THREAD

**Issue by muskit** [mahrtayyab/tweety#65]
When calling `tweet_details` on this particular retweet, I get the following error:

```
>>> app.tweet_detail('https://twitter.com/petra_gurin/status/1690719305971994624')
Traceback (most recent call last):
  File ".../.venv/lib/python3.11/site-packages/tweety/bot.py", line 233, in tweet_detail
    if raw_tweet['rest_id'] == str(tweetId):
       ~~~~~~~~~^^^^^^^^^^^
KeyError: 'rest_id'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../.venv/lib/python3.11/site-packages/tweety/bot.py", line 237, in tweet_detail
    raise InvalidTweetIdentifier(144, "StatusNotFound", r)
tweety.exceptions_.InvalidTweetIdentifier: The Tweet Identifier is Invalid
```

Probably related issue: when running into that retweet via `get_tweets`, no information other than tweet ID is present. Crucially missing is `retweeted_tweet`.
```
>>> BAD_TWEET
Tweet(id=1690719305971994624, author=None, created_on=None)
>>> BAD_TWEET.retweeted_tweet
>>> 
```

Using the version from `pip`

**Comment by muskit** [comment]
Think it might have something to do with the retweeted tweet, as `tweet_detail` fails on it.
https://twitter.com/014_shirasudon/status/1690716354985566208
```
>>> app.tweet_detail("https://twitter.com/014_shirasudon/status/1690716354985566208")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../.venv/lib/python3.11/site-packages/tweety/bot.py", line 226, in tweet_detail
    r = self.request.get_tweet_detail(tweetId)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.venv/lib/python3.11/site-packages/tweety/http.py", line 123, in get_tweet_detail
    response = self.__get_response__(**self.__builder.tweet_detail(tweetId, cursor))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.venv/lib/python3.11/site-packages/tweety/http.py", line 48, in __get_response__
    return GenericError(
           ^^^^^^^^^^^^^
  File ".../.venv/lib/python3.11/site-packages/tweety/types/n_types.py", line 60, in __init__
    self._raise_exception()
  File ".../.venv/lib/python3.11/site-packages/tweety/types/n_types.py", line 64, in _raise_exception
    raise self.EXCEPTIONS[self.error_code](
tweety.exceptions_.InvalidTweetIdentifier: _Missing: No status found with that ID.

**Comment by mahrtayyab** [comment]
It will be fixed in next release , changes are from Twitter Side

**Comment by muskit** [comment]
After looking into it, seems like the issue occurs on any tweet with replies restricted. I partially fixed it in #68, might finish it up if you would like. I'll probably also clean up the code to match your styling (particularly with existence of key in a dict) if you want me to continue with it
